src/Lottery.js

```diff
@@ -0,0 +1,192 @@
+ * @param {string} key - The key to use in the storage area.
+ * @param {*} value - The value to store in the storage area for the key.
+ */
+async function set(key, value) {
+    await browser.storage.local.set({ [key]: value });
+}
+
+/**
+ * Set the browser action popup to the survey's no prompt page.
+ * @private
+ */
+function setPopupToNoPromptPage() {
+    browser.browserAction.setIcon({path: "src/browser-action-assets/icon.png"});
+    browser.browserAction.setPopup({
+        popup: browser.runtime.getURL("src/browser-action-assets/userSurvey.popupNoPrompt.html")
+    });
+}
+
+/**
+ * Check if user is a lottery winner and if so, notify.
+ * @param {Object} rally - The Mozilla Rally object for this study, initialized in background.js
+ * @param {Boolean} is_dev_mode - Changes storage location based on whether this is dev mode or not
+ */
+export async function checkWinner ({
+  rally: rally,
+  is_dev_mode: is_dev_mode
+}) {
+  // If this module has already been initialized, don't do it again
+  if (initialized) {
+    console.log('already')
+    return
+  }
+  initialized = true
+
+  // Set Objects and Parameters
+  const surveyUserID = await webScience.userSurvey.getSurveyId() // the User ID
+  console.log("The UserID is " + surveyUserID)
+
+  var lottery_interval = 1440; // how often lottery results checked (in minutes)
+  var lottery_delay = 0.5; // when the lottery results are first checked (in minutes)
+  console.log("Lottery will be checked every " + lottery_interval + " minutes")
+  var lotteryAlarm = "lottery-alarm"; // name of lottery alarm
+  var winnerNotification = "winner-notification"; // name of winner notification
+  browser.alarms.create(lotteryAlarm, {periodInMinutes: lottery_interval, delayInMinutes: lottery_delay}); // create alarm for checking lottery results
+
+  browser.alarms.onAlarm.addListener(function(alarm) { // check lottery results when alarm goes off
+    // Run Fetch
+    fetchResults(surveyUserID).then(lotteryresults => {
+      if (lotteryresults['statusCode'] !== 200) { // there was an error
+        console.log("There was an error!")
+      } else if (lotteryresults['data']['isWinner']) { // the user is a winner
+        console.log(surveyUserID + " is a winner!")
+
+        // Set Redeem Code
+        const redeemCode = lotteryresults['data']['giftCode']
+        set("redeemCode", redeemCode)
+        console.log("The redeem code is " + redeemCode)
+        
+        // Set Redeem Status
+        const redeemed = lotteryresults['data']['redeemed']
+
+        // Send Alerts
+        if (!redeemed) { // user has not yet redeemed code
+          // Set up Notification alert
+          browser.notifications.create(winnerNotification, {
+            "type": "basic",
+            "title": "Beyond the Paywall Lottery",
+            "message": "Congratulations, you have won! Click the \"Beyond the Paywall\" extension icon for more details.",
+          });
+          console.log("Notification launched");
+
```

Comment:
Have you considered launching a new tab here instead of the notification? If notifications are off and/or if the extension icon is hidden, discoverability is very difficult. The new tab would also give you an opportunity to make the "winner" page look more official.

Reply:
Good suggestion, but we'll just keep the notification for now. It might be more jarring for a user to suddenly have a new tab pop open while browsing and it would take more time on our end to design a new page for this.